src/main/scala/subsystem/HasTiles.scala

```diff
@@ -10,13 +10,18 @@ import freechips.rocketchip.devices.tilelink.{BasicBusBlocker, BasicBusBlockerPa
 import freechips.rocketchip.diplomacy._
 import freechips.rocketchip.diplomaticobjectmodel.logicaltree.{LogicalModuleTree}
 import freechips.rocketchip.interrupts._
-import freechips.rocketchip.tile.{BaseTile, LookupByHartIdImpl, TileParams, HasExternallyDrivenTileConstants, InstantiableTileParams, PriorityMuxHartIdFromSeq}
+import freechips.rocketchip.tile.{BaseTile, LookupByHartIdImpl, TileParams, InstantiableTileParams, MaxHartIdBits}
 import freechips.rocketchip.tilelink._
 import freechips.rocketchip.util._
 
 /** Entry point for Config-uring the presence of Tiles */
 case class TilesLocated(loc: HierarchicalLocation) extends Field[Seq[CanAttachTile]](Nil)
 
+/** Whether a global hart id prefix is going to be dynamically applied to the static tile hart ids */
+case object HartPrefixKey extends Field[Boolean](false)
+case object SubsystemExternalHartIdWidthKey extends Field[Option[Int]](None)
```

Comment:
I dont get this set of knobs. HartPrefixKey is binary yes or no, so if I set it to yes, how does SubsystemExternalHartIdWidthKey get used? How do I know how wide the prefix is vs the static part?

Could we add a comment with a simple ASCII art something like this:

```
----------------------------------------------------------------------
| <---dynamic_hart_prefix_bits ---> | <---- static_tile_id_bits ---> |
---------------------------------------------------------------------
dynamic_hart_prefix_bits + static_tile_id_bits-1                     0
                     HartId
```
----------------------------------------------------------------
| <---dynamic hart prefix bits ---> | <---- statid tile id bits ---> |
----------

Reply:
I now understand that a "prefix" is really any number of "HartId" inputs of the same width `SubsystemExternalHartIdWidth`, and you can connect any number of them via the tileHartIdNexusNode

I'll resolve this comment to rescind my request for ASCII art and see if there are places I should suggest comments that would have made this more obvious to me.